plugin.video.iranintl/default.py

```diff
@@ -0,0 +1,14 @@
+import xbmc, xbmcgui, os, xbmcaddon, xbmcplugin
+
+__settings__ = xbmcaddon.Addon()
+home = __settings__.getAddonInfo('path')
+addon_handle = int(sys.argv[1])
+icon = xbmc.translatePath(os.path.join(home, 'icon.png'))
+li = xbmcgui.ListItem ('Live Stream')
+li.setArt({ 'thumb': icon })
+li.setInfo('video', {'plot':'Iran International TV Live Stream'})
+
+liveVideoUrl = "https://live.playstop.live/LS-63503-1/index.m3u8"
```

Comment:
Could you provide us the site where you got this URL from? Just to check the source is available and free for the public?

Reply:
Sure! https://iranintl.com/live

Reply:
Seems like that one is pointing to https://live.livetvstream.co.uk/LS-63503-3/index.m3u8

Reply:
You're right, that's the current behavior. I'll update the URL!

    function pickOrigin()
    {
        return "https://live.livetvstream.co.uk/LS-63503-3/index.m3u8";
        if(Math.random() < EXPERIMENT_ORIGIN_FRAC) {
            return "https://dev-live.livetvstream.co.uk/LS-63503-4/index.m3u8";
        }
        return "https://live.playstop.me/1816184091/index.m3u8";
    }

Reply:
I would say you fetch https://iranintl.com/live and extract the URL from it. That way you will always be right?

Reply:
Yes, I generally prefer that approach too :) Need to execute JS in this case though as parsing alone won't do. Is there a phantom.js module I can import? Some older documentation suggests that I might have to provide binaries for every platform?

Reply:
You can just fetch the js url data and parse it using a regular expression?

Reply:
Problem is, all I get is 3 separate 'return "https://...";' lines as per the above JS function. They'll be all equally 'valid'. I can make the assumption that the first one is the one I'm looking for which may or may not be true depending on the next change they make.

Reply:
I guess that's arguably still a slight improvement over a hard-coded URL? Thoughts on using the first m3u8 URL I find in the page?

Reply:
The pickOrigin now always returns "https://live.livetvstream.co.uk/LS-63503-3/index.m3u8" so I would take the first of the 3 matches

Reply:
Done!